src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/OpenSslX509CertificateReader.cs

```diff
@@ -0,0 +1,572 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Formats.Asn1;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Asn1;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal sealed class OpenSslX509CertificateReader : ICertificatePal
```

Comment:
Please avoid using the name OpenSsl in any Android-specific types.   If it's needed because of sharing too much code with the Unix build as a temporary measure, please name the .cs file what the type is expected to be in the end (e.g. AndroidX509CertificateReader.cs).

Reply:
Yeah, it is the same class name right now is because it is currently sharing a lot with the Unix build - will rename the .cs file.